src/URI.js

```diff
@@ -1863,6 +1863,7 @@
 
     if (resolved.path().charAt(0) !== '/') {
       basedir = base.directory();
+      basedir = basedir ? basedir : base.path().indexOf('/') === 0 ? '/' : '';
```

Comment:
Yes, I know. It somehow seems that there should be a cleaner way to do this... :-) But I couldn't find it (admittedly, 

As for changing the ?: to ||&& - that's a question of style. My code is more "say what you mean", yours is more concise. I could go either way.